### DISCUSSION TITLE
How to set up bot owners or admins

### DISCUSSION THREAD

**Discussion by Diego-Schmidt** [vasani-arpit/WBOT#235] [Q&A]
Hello, Is there any example on how to set up bot owners to respond to "admin" commands?

You get the user id from message.chatId.user ?
Can you do something like

$owner = '65465456465';
$getuser = ¿?¿?¿ message.chatId.user;

 if($message == '/restart'  && $getuser == $owner) 
{ do something }
else {
echo 'You can't use that command';
}

**Comment by vasani-arpit** [comment]
No. There is not. but it can be developed. Can you explain your entire use case?

**Reply by Diego-Schmidt** [reply]
In my case i was thinking to use something like that to have a bot command to activate or deactivate a "gloriafood shop" button on a website. The scenario is to avoid people to use the website shop button in case of "system crash" or "product delivery problems" with the platform. Something like "/shopoffline /shoponline" command to quickly change the state of the shop button. But there are a lot of things to do with restricted commands.

**Reply by vasani-arpit** [reply]
### How I would achieve this?
1. add your function inject.js or in index.js (keep in mind that inject.js is browser based and index.js is node based)  
2. define a character which will trigger a function which is '/' in your example. I suggest you to go with other which is used less
3. now whenever the charactor from 2nd step comes in message, look or remaining message and check if there is a function available with that name. If yes then execute it. if no then reply with no function found

Put all this functions in one file so that you can keep on adding the function without updating index or inject files. 

### Is this part of WBOT?
I don't think so. At the moment there are other items which needs fixing and there are other features which needs to be implemented first. so currently can't add support for this in WBOT. Maybe in the future.

**Comment by BroodPitt** [comment]
Ah yes it is possible


$x = file_get_contents("php://input");
$x = json_decode($x);
$message = $x->text;
$msg_typ = $x->type;
$sender= $x->user; <--------- add that to your PHP file... and trigger/check from that.
